fish-rust/src/builtins/wait.rs

```diff
@@ -177,15 +176,16 @@ pub fn wait(parser: &Parser, streams: &mut IoStreams, argv: &mut [&wstr]) -> Opt
 
     // Get the list of wait handles for our waiting.
     let mut wait_handles: Vec<WaitHandleRef> = Vec::new();
-    for i in w.woptind..argc {
-        if iswnumeric(argv[i]) {
+    let optind = w.woptind;
+    for item in argv.iter().take(argc).skip(optind) {
```

Comment:
`argc` is not the length, so the equivalent would be `take(argc - optind)`.

better to write `for arg in &argv[optind..argc]`. Clippy not smart enough

Reply:
Blergh.

I *think* I stumbled over that one before? I probably should trust clippy less.